backend/eventloop/application_event_loop/application_event_runner_syncruns.go

```diff
@@ -33,20 +33,93 @@ func (action *applicationEventLoopRunner_Action) applicationEventRunner_handleSy
 	// Handle all GitOpsDeploymentSyncRun related events
 	err := action.applicationEventRunner_handleSyncRunModifiedInternal(ctx, dbQueries)
 
-	// TODO: GITOPSRVCE-44: Get the SyncRunModified object from k8s, so we can update it if necessary
-
-	// TODO: GITOPSRVCE-44: Add the ErrorOccurred condition to the syncRunModified
-
-	// TODO: GITOPSRVCE-44: If no user error (just dev error), then output generic error occurred
+	syncRunCR, clientErr := getGitOpsDeploymentSyncRun(ctx, action.workspaceClient, action.eventResourceName, action.eventResourceNamespace)
+	if clientErr != nil {
+		if !apierr.IsNotFound(clientErr) {
+			return fmt.Errorf("unable to get GitOpsDeploymentSyncRun: %v", err)
+		}
+		return nil
+	}
 
+	conditionType := managedgitopsv1alpha1.GitOpsDeploymentSyncRunConditionErrorOccurred
 	if err != nil {
+		// set the condition only for user errors
```

Comment:
In the case where an error occurred, but it is a dev-only error, I think we should still set the Error Ocurred condition, BUT we should use a generic error message like 'an internal error occurred'.

This is what I did for `GitOpsDeployment`s:
https://github.com/redhat-appstudio/managed-gitops/blob/6197011ff97aa22cefb45e9a96a9022ab8243d2d/backend/eventloop/application_event_loop/application_event_runner_deployments.go#L910

It's not perfect, because it presents the user with an error that they have no way to resolve (they should probably contact support), but it does at least tell them that something is wrong. WDYT?